### PR TITLE
refactor registry.zookeeper & define config center module interface

### PR DESCRIPTION
**Pull request by hxmhlt**
**What this PR does**:
refactor registry.zookeeper & define config center module interface

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

```release-note

```

### CODE REVIEW

**Comment by ralf0131** [comment]
Hi, could you please resolve the conflict?

File: config_center/dynamic_configuration.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package config_center
+
+import (
+	"github.com/apache/dubbo-go/remoting"
```
Comment (fangyincheng) [inline config_center/dynamic_configuration.go]: formatting pkg

File: config_center/zookeeper/dynamic_configuration.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zookeeper
+
+import (
+	"github.com/apache/dubbo-go/remoting"
```
Comment (fangyincheng) [inline config_center/zookeeper/dynamic_configuration.go]: format pkg

File: config_center/zookeeper/dynamic_configuration.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+)
+import (
+	"github.com/apache/dubbo-go/common"
+	"github.com/apache/dubbo-go/common/constant"
+	"github.com/apache/dubbo-go/common/logger"
+	"github.com/apache/dubbo-go/config_center"
+	"github.com/apache/dubbo-go/remoting/zookeeper"
+)
+
+const ZkClient = "zk config_center"
```
Comment (fangyincheng) [inline config_center/zookeeper/dynamic_configuration.go]: const should be uppercase.

File: registry/event.go [hunk 4]
```diff
@@ -19,6 +19,7 @@ package registry
 
 import (
 	"fmt"
+	"github.com/apache/dubbo-go/remoting"
```
Comment (fangyincheng) [inline registry/event.go]: fmt pkg

File: registry/zookeeper/listener.go [hunk 5]
```diff
@@ -19,295 +19,64 @@ package zookeeper
 
 import (
 	"context"
-	"fmt"
-	"path"
-	"sync"
-	"time"
+	"github.com/apache/dubbo-go/remoting"
```
Comment (fangyincheng) [inline registry/zookeeper/listener.go]: fmt pkg

File: registry/zookeeper/listener.go [hunk 6]
```diff
@@ -19,295 +19,64 @@ package zookeeper
...
-
 import (
 	"github.com/apache/dubbo-go/common"
 	"github.com/apache/dubbo-go/common/logger"
 	"github.com/apache/dubbo-go/registry"
+	zk "github.com/apache/dubbo-go/remoting/zookeeper"
 )
 
-const (
-	MaxFailTimes = 15
```
Comment (fangyincheng) [inline registry/zookeeper/listener.go]: const should be uppercase.

File: registry/zookeeper/registry.go [hunk 7]
```diff
@@ -40,13 +40,13 @@ import (
...
 	"github.com/apache/dubbo-go/version"
 )
 
 const (
-	defaultTimeout    = int64(10e9)
-	RegistryZkClient  = "zk registry"
-	RegistryConnDelay = 3
+	defaultTimeout   = int64(10e9)
+	RegistryZkClient = "zk registry"
 )
```
Comment (fangyincheng) [inline registry/zookeeper/registry.go]: const should be uppercase.

File: remoting/zookeeper/listener.go [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zookeeper
+
+import (
+	"github.com/apache/dubbo-go/remoting"
```
Comment (fangyincheng) [inline remoting/zookeeper/listener.go]: fmt pkg

**Comment by hxmhlt** [comment]
@ralf0131 Resolved.

**Comment by AlexStocks** [comment]
LGTM.

**Pull request merged**